scripts/MergeAnalyzerMetadata.ps1

```diff
@@ -75,20 +48,20 @@ try {
 
     Copy-Item -Path $DiagToolPath -Destination $tempDir
 
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.Analyzers.Extra.dll
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.Analyzers.Local.dll
-    & .\DiagConfig.exe $Diags analyzer merge StyleCop.Analyzers.dll
-    & .\DiagConfig.exe $Diags analyzer merge SonarAnalyzer.CSharp.dll
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.VisualStudio.Threading.Analyzers.dll
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.VisualStudio.Threading.Analyzers.CSharp.dll
-    & .\DiagConfig.exe $Diags analyzer merge xunit.analyzers.dll
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.CodeAnalysis.CodeStyle.dll
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.CodeAnalysis.CSharp.CodeStyle.dll
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.CodeAnalysis.NetAnalyzers.dll
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll
-    & .\DiagConfig.exe $Diags analyzer merge ILlink.RoslynAnalyzer.dll
-#    & .\DiagConfig.exe $Diags analyzer merge Microsoft.AspNetCore.App.Analyzers.dll
-    & .\DiagConfig.exe $Diags analyzer merge Microsoft.AspNetCore.Components.Analyzers.dll
+    & dotnet exec .\DiagConfig.dll $Diags analyzer merge Microsoft.Analyzers.Extra.dll
```

Comment:
Why change this stuff here?

Reply:
When calling it via `.\DiagConfig.exe` the project will always try to use the globally installed dotnet (meaning the one in `C:\Program Files\dotnet\`, but that won't always be the same version of the SDK that was used to build the tool. Since we are building the tool a few lines above using `dotnet` (whichever path wins for that), here I'm changing it so that we also just use `dotnet` for the execution so we can guarantee that it uses the same SDK for running than the one it used for building.

I hit this locally because I didn't have the version listed in global.json, so I temporarily added the local path of the restored dotnet tool to my path by running `$env:PATH="C:\Users\joperezr\git\extensions\.dotnet;" + $env:PATH`. After doing this, running the script fails locally as I'm building the tool using the locally restored SDK, but it is trying to run with the globally installed one which doesn't match. After my changes things work as expected.

Reply:
I guess I'll need to do the same thing to the MakeEditorConfigs.ps1 and MakeApiBaselines.ps1...

Reply:
It would surely help making things consistent 😃

Reply:
This should really be:
```suggestion
    & ./.dotnet/dotnet exec ./DiagConfig.dll $Diags analyzer merge Microsoft.Analyzers.Extra.dll
```

Note, all paths must be in POSIX so the script is runnable on Linux as well.